### ISSUE TITLE
Support phase enumeration has changed

### ISSUE THREAD

**Issue by tillig** [jonstodle/DotNetSdkHelpers#18]
It appears just today the [contents of the `releases.json`](https://raw.githubusercontent.com/dotnet/core/master/release-notes/releases-index.json) has updated so [the new set of support phases are](https://github.com/dotnet/core/blob/e349085c452b56642fd82bf0413d5b3280a1382c/release-policies.md):

- preview
- go-live
- active
- maintenance
- eol

The field indicating LTS vs STS (standard term support) has moved to a `release-type` field.

Basically, the data model (and terminology) around all the releases has changed.

I'm already in the process of setting up a PR [to convert the build to .NET 6](https://github.com/jonstodle/DotNetSdkHelpers/tree/feature/net6) with analyzers, etc. so I can include this in that (since I can't test it without that update anyway).

**Comment by tillig** [comment]
It appears this is going to affect the CLI commands since some of the options don't make sense in this setup anymore.

- `dotnet-sdk download {lts|current|preview}`
- `dotnet-sdk list --available --lts-only`
- `dotnet-sdk list --available --preview-only`

Splitting the release type (LTS/STS) from the support phase (preview/active) means something like `current` may equate to "most recent `active`" or something. However, trying to boil down the options like that may not make sense if people want more power over filtering

For `dotnet-sdk list` I'm thinking about switching this to take new multi-valued parameters that provide more robust filtering.

- `-r | --release-type`: Filter by release type
- `-s | --support-phase`: Filter by support phase

For example:

- `dotnet-sdk list -a -r lts`: List the available LTS releases.
- `dotnet-sdk list -a -s active -s maintenance`: List the available "active" and "maintenance" releases.
- `dotnet-sdk list -a -r lts -s active`: List the available LTS releases considered active.

Likely the command output will also need to be updated to differentiate support and release.

The `dotnet-sdk download` is a little harder, but we could switch it to just be the `support-phase` instead of the `release-type` so...

- `dotnet-sdk download active`: Get the latest active release.
- `dotnet-sdk download preview`: Get the latest preview release.
- `dotnet-sdk download go-live`: Get the latest RC (go-live) release.
- `dotnet-sdk download 6.0`: Get the latest in the 6.0 release line.

That limits the ability to say "get me the most recent LTS version," but, based on this new breakdown, the latest LTS version may be in preview, so you wouldn't get what you _think_ you're getting if you're unaware of the separation here.
